### PR TITLE
Included additional tests for server.go

### PR DESCRIPTION
**Pull request by neilnaveen**
### 1. Why is this pull request needed and what does it do?
- Included additional tests for server.go

Signed-off-by: neil <42328488+neilnaveen@users.noreply.github.com>

### CODE REVIEW

**Comment by neilnaveen** [comment]
@yongtang can you please review?

File: core/dnsserver/server_test.go [hunk 1]
```diff
@@ -2,15 +2,33 @@ package dnsserver
...
+	"reflect"
+	"sync"
 	"testing"
+	"time"
 
 	"github.com/coredns/coredns/plugin"
 	"github.com/coredns/coredns/plugin/pkg/log"
+	"github.com/coredns/coredns/plugin/pkg/trace"
 	"github.com/coredns/coredns/plugin/test"
-
```
Comment (chrisohaver) [inline core/dnsserver/server_test.go:10]: removing this line will cause import order/grouping tests to fail.

File: core/dnsserver/server_test.go [hunk 2]
```diff
@@ -93,3 +111,149 @@ func BenchmarkCoreServeDNS(b *testing.B) {
...
+				classChaos:   tt.fields.classChaos,
+			}
+			if got := s.WrapListener(tt.args.ln); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Server.WrapListener() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestServer_Address(t *testing.T) {
```
Comment (chrisohaver) [inline core/dnsserver/server_test.go:193]: `Server.Address` is a single line function that returns simply Addr field.
It's too simple to warrant a test of this bulk.

File: core/dnsserver/server_test.go [hunk 3]
```diff
@@ -93,3 +111,149 @@ func BenchmarkCoreServeDNS(b *testing.B) {
...
+				dnsWg:        sync.WaitGroup{},
+				graceTimeout: time.Second,
+				trace:        nil,
+				debug:        false,
+				classChaos:   false,
+			},
+			want: "",
+		},
+		{
+			name: "nil case",
```
Comment (chrisohaver) [inline core/dnsserver/server_test.go:226]: "nil case" doesn't describe this case. ":53" isn't nil.

**Review by chrisohaver — CHANGES_REQUESTED** [review]
Neither of the functions tested warrant such complex test cases.
They are both one line functions that simply return a single struct field.

Were these tests created by a unit test code generator?

**Pull request closed without merging**